src/App.tsx

```diff
@@ -68,12 +68,20 @@ const App: React.FC = () => {
         }
     }, [])
 
+    const initData = async () => {
+        await initialActiveShopListData()
+        await fetchCategories()
+    }
+
     useEffect(() => {
         console.log('user id')
         console.log(user)
         if (user !== null) {
-            initialActiveShopListData()
-            fetchCategories()
+            // initData()
+            fetchCategories().then(() => {
+                initialActiveShopListData()
+                setInit(false)
+            })
```

Comment:
Does async not work here? I saw that you created initData method but not used it.

Reply:
Yes. Actually, it still does not work. The problem is when you come from the login/register page. The application loads the items before the category which causes the errors. The thing is, If I listen for categories in the itemsPage to wait before loading items, it will rerender the itemsPage every time I update a single category name...

Reply:
Could this be solved by extracting the getItems method on itemsPage into own hook? So then you load initial data in the App,tsx instead of ItemsPage. I have this task on the backlog (creating that hook, since right now it loads the items data on every mount and unmount of ItemsPage).

Reply:
I ended up using a loaded state on categories. Yes, but in your case, you will load the items even if you are not in the items page?

Reply:
Yes, when the user logs in it would load the items only once. Afterwards, if you would need to sync local items with the DB items, you can just use that hook. We could also only load items if the user goes to /items path, so for example if the user first goes to /history items wouldn't be loaded.

Reply:
I add a useFetchItems hook and handle the fetch where needed ( add/edit sidebar ) and I also add the possibility to add the item to the shoplist when in ShowItemSidebar. Everything seems to works fine. At least, at first sight 🥇